File: Leetcode/30.cpp
#include<vector>
#include<iostream>
#include<string>
using namespace std;
class Solution {
public:
    vector<int> findSubstring(string s, vector<string>& words) {
        vector<int> arr;
        int totalSub = words.size() * words[0].length();
        if (totalSub>s.length()){
            return arr;
        }
        else{
            for(int i=0;i<s.length()-totalSub+1;i++){
                string st = "";
                for(int j = i;j<i+totalSub;j++){
                    st+=s[j];
                }
                vector<string> tempList;
                for(int j = 0; j<st.length();j+=words[0].length()){
                    string x = "";
                    for(int k = j;k<j+words[0].length();k++){
                        x+=st[k];
                    }
                    tempList.push_back(x);
                }
                bool available = true;
                vector<string> tempArr;
                for(int j=0;j<words.size();j++){
                    tempArr.push_back(words[j]);
                }
                for(int j=0;j<tempList.size();j++){
                    bool inside = false;
                    for(int k = 0;k<tempArr.size();k++){
                        if(tempArr[k] == tempList[j]){
                            inside = true;
                        }
                    }
                    if(inside == false){
                        available = false;
                        break;
                    }
                    else{
                        vector<string> ::iterator it;
                        it = find(tempArr.begin(),tempArr.end(), tempList[j]);
                        tempArr.erase(it);
                    }
                }
                if(available){
                    arr.push_back(i);
                }
            }
            return arr;
        }
        return arr;
    }
};

File: Leetcode/30.py
class Solution:
    def findSubstring(self, s: str, words: List[str]) -> List[int]:
        totalSub = len(words)* len(words[0])
        if totalSub>len(s):
            return []
        else:
            arr = list()
            for i in range(len(s)-totalSub+1):
                st = s[i:i+totalSub]
                tempList = list()
                for j in range(0,len(st),len(words[0])):
                    tempList.append(st[j:j+len(words[0])])
                available = True
                tempArr = words.copy()
                for ele in tempList:
                    if ele not in tempArr:
                        available = False
                        break
                    else:
                        tempArr.remove(ele)
                if available:
                    arr.append(i)
            return arr